airbyte-integrations/connectors/destination-pinecone/integration_tests/spec.json

```diff
@@ -24,6 +24,12 @@
             "title": "Index",
             "description": "Pinecone index in your project to load data into",
             "type": "string"
+          },
+          "store_text": {
+            "title": "Store text",
+            "description": "Store the text that gets embedded along with the vector and the metadata in the Pinecone index under the key `text`. If set to false, only the vector and the metadata will be stored.",
```

Comment:
I'm not clear on the use case for this... Can you explain a bit more?

Reply:
Thanks for your help to understand this paradigm. What do you think of this approach:

1. Invert the paradigm to "Don't store raw text" as an opt-in field, disabled by default.
2. Put this new setting behind "advanced" options if we can. (Perhaps we're already doing this but wanted to call it out.

Rationale would be:
1. This is really an edge case for purposes of large datasets and/or data privacy concerns.
2. Most users don't need to use this option and shouldn't use it if they are unsure.
3. If someone disables the text field, it will break the normal RAG use cases, in which the RAG flow expects to pull down the raw text as a result of the vector scan. 

I think phrasing in the inverse, default to disabled, slightly discourages this from being selected - unless the user knows they want this behavior and searches for it.

Wdyt?

I think also - while the nuance about whether stored in "metadata" specifically should probably be left out of the setting name. While it was helpful to me in better understanding the internals, it could be confusing to users who don't know the default behavior of Pinecone, such as when setting up for the first time. (We also have a separate "metadata" field input in our settings, and we don't want to conflate what we are _asking_ for as metadata, vs what the vector stores as metadata.

I think these nuances can be explained in the tooltip and in the docs (thanks for linking this field to them, btw 🙏). Lastly, I do think a generic "Don't store raw text" could be applicable to other destinations equally - and nicely abstract the internal implementation details for the purpose of accessibility.

Wdyt?

Reply:
Good points, I will close this PR and re-open as an extension of the vector DB CDK